src/font.js

```diff
@@ -87,11 +87,43 @@ function Font(options) {
         this.ascender = options.ascender;
         this.descender = options.descender;
         this.createdTimestamp = options.createdTimestamp;
+        this.slope = options.slope;
+        this.italicAngle = options.italicAngle;
+
+        let selection = 0;
+        if (this.italicAngle < 0) {
+            selection |= this.fsSelectionValues.ITALIC;
+        } else if (this.italicAngle > 0) {
+            selection |= this.fsSelectionValues.OBLIQUE;
+        }
+        if (this.weightClass >= 600) {
+            selection |= this.fsSelectionValues.BOLD;
+        }
+        if (selection == 0) {
+            selection = this.fsSelectionValues.REGULAR;
+        }
+        if (options.fsSelection) {
```

Comment:
If `options.fsSelection` is set, all the other checks above are not necessary, so we should check that first and only perform the other checks otherwise.

Reply:
I was able to address 3 out of your 4 comments:
* I added an initial check to see if `fsSelection` was specified, and only calculate it's value if it wasn't specified.
* Added a test to make sure Panose generates default values
* Added a test to make sure, in the case where `fsSelection` was not specified, that the auto-calculation happens 

The one thing I couldn't figure out is how to access `macStyle` to test it. It seems that, after a font is generated, `font.tables.head.macStyle` does not exist. I tried to add a test for this, but it failed (due to it not existing)

Do you know how to access `macStyle` after the font is generated, so I can add a test for it?

Reply:
When you create a new font object, there is no head table being created automatically. What we want to test here is if the provided data is reflected in the generated table data when saving the font, i.e. when `toTables()` is called (which happens internally when downloading the font).

```js
// after creating your font object
font.tables.head = { macStyle: 1 };
const head = font.toTables().tables.find(table => table.tableName === 'head');
assert.equal(head.macStyle, 1);
```

Please also test the panose values when they are specified, right now just the default value fallback is tested. Thanks!

Reply:
Got it - just did some more investigation and I found the `toTables` method. I also looked through the OTF spec to see what the `macStyle` bits should be, and discovered they should match the `fsSelection` bits... so I updated how `macStyle` is calculated.

Reply:
All right - just updated the tests and added a test where panose values are specified.

... anything else? Let me know!

Reply:
Just one thing: When the spec says the `fsSelection` and `macStyle` bits should "agree", that doesn't mean that they should be exactly the same. Just the corresponding meaning must match (i.e. if `fsSelection` says "bold italic", `macStyle` should say "bold italic" as well). But the bits used actually differ, see
https://learn.microsoft.com/en-us/typography/opentype/spec/os2#fsselection (here you can even see the bits used for `macStyle` in the table)
and https://learn.microsoft.com/en-us/typography/opentype/spec/head
> Bit 0: Bold (if set to 1);
> Bit 1: Italic (if set to 1)
> Bit 2: Underline (if set to 1)
> Bit 3: Outline (if set to 1)
> Bit 4: Shadow (if set to 1)
> Bit 5: Condensed (if set to 1)
> Bit 6: Extended (if set to 1)
> Bits 7–15: Reserved (set to 0).
It would be best to add these values in analogy to `Font.prototype.fsSelectionValues = {...}` in `font.js` and make use of them like that.

Reply:
Oh, good catch - I misinterpreted the meaining.  I'll make sure the `macStyle` bits are accurate based on the reference you linked to above.

Reply:
Bit names added as you suggested, and now used while calculating `macStyle`